src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -0,0 +1,574 @@
+            }
+        });
+
+        // We do not have a color ramp to support light mode... so, force dark.
+        RequestedTheme(ElementTheme::Dark);
+    }
+
+    void ExtensionPalette::AIKeyAndEndpoint(const winrt::hstring& endpoint, const winrt::hstring& key)
+    {
+        _AIEndpoint = endpoint;
+        _AIKey = key;
+        _httpClient = winrt::Windows::Web::Http::HttpClient{};
+        _httpClient.DefaultRequestHeaders().Accept().TryParseAdd(L"application/json");
+        _httpClient.DefaultRequestHeaders().Append(L"api-key", _AIKey);
+    }
+
+    void ExtensionPalette::IconPath(const winrt::hstring& iconPath)
+    {
+        // We don't need to store the path - just create the icon and set it,
+        // Xaml will get the change notification
+        ResolvedIcon(winrt::Microsoft::Terminal::Settings::Model::IconPathConverter::IconWUX(iconPath));
+    }
+
+    winrt::fire_and_forget ExtensionPalette::_getSuggestions(const winrt::hstring& prompt, const winrt::hstring& currentLocalTime)
+    {
+        const auto userMessage = winrt::make<ChatMessage>(prompt, true, false);
+        std::vector<IInspectable> userMessageVector{ userMessage };
+        const auto userGroupedMessages = winrt::make<GroupedChatMessages>(currentLocalTime, true, _ProfileName, winrt::single_threaded_vector(std::move(userMessageVector)));
+        _messages.Append(userGroupedMessages);
+        _queryBox().Text(L"");
+
+        TraceLoggingWrite(
+            g_hQueryExtensionProvider,
+            "AIQuerySent",
+            TraceLoggingDescription("Event emitted when the user makes a query"),
+            TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA),
+            TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+
+        // request the latest LLM key and endpoint
+        _AIKeyAndEndpointRequestedHandlers(nullptr, nullptr);
+
+        // if the AI key and endpoint is still empty, tell the user to fill them out in settings
+        if (_AIKey.empty() || _AIEndpoint.empty())
+        {
+            _splitResponseAndAddToChatHelper(RS_(L"CouldNotFindKeyErrorMessage"));
+
+            TraceLoggingWrite(
+                g_hQueryExtensionProvider,
+                "AIResponseReceived",
+                TraceLoggingDescription("Event emitted when the user receives a response to their query"),
+                TraceLoggingBoolean(false, "ResponseReceivedFromAI", "True if the response came from the AI, false if the response was generated in Terminal or was a server error"),
+                TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA),
+                TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+
+            co_return;
+        }
+        else if (!std::regex_search(_AIEndpoint.c_str(), azureOpenAIEndpointRegex))
+        {
+            _splitResponseAndAddToChatHelper(RS_(L"InvalidEndpointMessage"));
+
+            TraceLoggingWrite(
+                g_hQueryExtensionProvider,
+                "AIResponseReceived",
```

Comment:
(same with this one)

Reply:
(resolving this to keep the discussion in one place)